src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,44 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
+
+  die() {
+    Animal.alive = Animal.alive.filter((animal) => animal !== this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+  }
+
+  bite(animal) {
+    if (animal instanceof Herbivore && !animal.hidden) {
+      animal.health -= 50;
+
+      if (animal.health <= 0) {
+        animal.die();
```

Comment:
[CODE PERFORMANCE] - You are currently calling the `die` method inside the `bite` method, which leads to filtering the `Animal.alive` array every time an animal's health drops below zero. This can be inefficient if multiple animals die in a sequence, as it would cause multiple iterations over the `Animal.alive` array. Consider filtering the `Animal.alive` array outside of the `bite` method after all bite actions are completed to improve performance. For example, you could have a separate method that checks and filters dead animals from the `Animal.alive` array at certain intervals or after a series of interactions.